jenkins/PerfCI_Chaos/03_PerfCI_Choas_Windows_VMs_Deployment/Jenkinsfile

```diff
@@ -112,30 +112,29 @@ END
             }
         }
 
-        stage('Prepare Google credential files') {
+        stage('Prepare Google Credential Files') {
             steps {
-                    script {
-                        // Assign the file path to the environment variable
-                        env.GOOGLE_DRIVE_CREDENTIALS = "${env.WORKSPACE}/${CREDENTIALS_FILE}"
-                        env.GOOGLE_DRIVE_TOKEN = "${env.WORKSPACE}/${TOKEN_FILE}"
-
-                        sh """
-                        echo "Checking if ${env.GOOGLE_DRIVE_CREDENTIALS} exists..."
-                        if [[ -f "${env.GOOGLE_DRIVE_CREDENTIALS}" ]]; then
-                            echo "File exists. Deleting it..."
-                            rm -f "${env.GOOGLE_DRIVE_CREDENTIALS}"
-                        fi
-                        echo "Checking if ${env.GOOGLE_DRIVE_TOKEN} exists..."
-                        if [[ -f "${env.GOOGLE_DRIVE_TOKEN}" ]]; then
-                            echo "File exists. Deleting it..."
-                            rm -f "${env.GOOGLE_DRIVE_TOKEN}"
-                        fi
+                script {
 
-                        echo "Copying secret file to the temporary local file..."
-                        cp "$GOOGLE_DRIVE_CREDENTIALS_FILE" "${env.GOOGLE_DRIVE_CREDENTIALS}"
-                        cp "$GOOGLE_DRIVE_TOKEN_FILE" "${env.GOOGLE_DRIVE_TOKEN}"
-                        """
+                    // Create a unique temp directory
+                    env.TEMP_DIR = sh(script: "mktemp -d -p '${env.WORKSPACE}'", returnStdout: true).trim()
+                    if (!env.TEMP_DIR) {
+                        error "Failed to create temporary directory!"
                     }
+                    echo "Temporary directory created: ${env.TEMP_DIR}"
+
+                    // Set up file paths
+                    env.GOOGLE_DRIVE_CREDENTIALS = "${env.TEMP_DIR}/${GOOGLE_CREDENTIALS_FILE}"
+                    env.GOOGLE_DRIVE_TOKEN = "${env.TEMP_DIR}/${GOOGLE_TOKEN_FILE}"
+
+                    // Prepare credential files
+                    sh """
+                        set -e
+                        cp '${GOOGLE_DRIVE_CREDENTIALS_FILE}' '${env.GOOGLE_DRIVE_CREDENTIALS}' || error "Failed to copy ${GOOGLE_DRIVE_CREDENTIALS_FILE}"
+                        cp '${GOOGLE_DRIVE_TOKEN_FILE}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to copy ${GOOGLE_DRIVE_TOKEN_FILE}"
+                        chmod 600 '${env.GOOGLE_DRIVE_CREDENTIALS}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to set permissions on credential files"
```

Comment:
Set the umask up front, to avoid even a very short vulnerability window and so that the second cp command fails the first one won't be left world readable.  Also, you don't need (or want) `set -e` here since you're handling errors yourself.

```
                        umask 77
                        cp '${GOOGLE_DRIVE_CREDENTIALS_FILE}' '${env.GOOGLE_DRIVE_CREDENTIALS}' || error "Failed to copy ${GOOGLE_DRIVE_CREDENTIALS_FILE}"
                       cp '${GOOGLE_DRIVE_TOKEN_FILE}' '${env.GOOGLE_DRIVE_TOKEN}' || error "Failed to copy ${GOOGLE_DRIVE_TOKEN_FILE}"
```

Reply:
fixed,
@RobertKrawitz thx for your useful comments